framework/utils.py

```diff
@@ -32,6 +32,32 @@ def is_internal_ip(ip):
     parsed_ip = IPAddress(ip)
     return parsed_ip.is_private
 
+def get_rank(user_rank, owtf_rank):
```

Comment:
As such, it would be better to separate the `get_rank` function from the ranking scale. By splitting the two, the ranking scale could be used somewhere else in the framework, in case another feature needs to translate the rankings.

Also, `get_rank` should check that the index is defined in the dict and default to a default value if it is not. For instance, something like:
```python
return RANKS.get(rank, 'Unknown')
```
The string `Unknown` should be defined somewhere else, or `Unranked` could be used. You get the idea.

 I would suggest specifying what is the return value in the docstring. Also, I am not sure to understand why the function should make the decision of which ranking to use. I believe that this should be done before calling it. Something like:
```python
# From the API
rank_str(max(user_rank, owtf_rank))
```

Finally, the name of the function could be a bit more specific in what it does with something like `rank_human_readable`, `rank_str`. From what I read, the function does not `get` anything, it simply translates a ranking from numerical to string value.